gdx/src/com/badlogic/gdx/maps/tiled/TmjMapLoader.java

```diff
@@ -0,0 +1,247 @@
+	 *
+	 * @param fileName the filename of the tile set
+	 * @param map the TiledMap to which the tile set will be added
+	 * @return the TiledMap with the loaded tile set */
+	public TiledMap loadCustomTileSet (String fileName, TiledMap map) {
+		return loadCustomTileSet(fileName, map, new TmjMapLoader.Parameters());
+	}
+
+	/** Loads a tile set from the given file. The file is resolved via the {@link FileHandleResolver} set in the constructor of
+	 * this class. By default it will resolve to an internal file.
+	 *
+	 * @param fileName the filename of the tile set
+	 * @return the TiledMap with the loaded tile set */
+	public TiledMap loadCustomTileSet (String fileName) {
+		TiledMap map = new TiledMap();
+		return loadCustomTileSet(fileName, new TiledMap(), new TmjMapLoader.Parameters());
+	}
+
+	/** Loads a tile set from the given file. The file is resolved via the {@link FileHandleResolver} set in the constructor of
+	 * this class. By default it will resolve to an internal file.
+	 *
+	 * @param fileName the filename of the tile set
+	 * @param map the TiledMap to which the tile set will be added
+	 * @param parameter specifies whether to use y-up, generate mip maps etc.
+	 * @return the TiledMap with the loaded tile set */
+	public TiledMap loadCustomTileSet (String fileName, TiledMap map, TmjMapLoader.Parameters parameter) {
+		FileHandle tmjFile = resolve(fileName);
+		JsonValue tileSet = json.parse(tmjFile);
+		ObjectMap<String, Texture> textures = new ObjectMap<>();
+		this.map = map;
+
+		final Array<FileHandle> textureFiles = getTileSetDependencyFileHandle(tmjFile, tileSet);
+		for (FileHandle textureFile : textureFiles) {
+			Texture texture = new Texture(textureFile, parameter.generateMipMaps);
+			texture.setFilter(parameter.textureMinFilter, parameter.textureMagFilter);
+			textures.put(textureFile.path(), texture);
+		}
+		loadTileSet(tileSet, tmjFile, new ImageResolver.DirectImageResolver(textures));
+		map.setOwnedResources(textures.values().toArray());
+
+		return map;
+	}
+
+	@Override
+	public void loadAsync (AssetManager manager, String fileName, FileHandle tmjFile, BaseTmjMapLoader.Parameters parameter) {
+		this.map = loadTiledMap(tmjFile, parameter, new ImageResolver.AssetManagerImageResolver(manager));
+	}
+
+	@Override
+	public TiledMap loadSync (AssetManager manager, String fileName, FileHandle file, BaseTmjMapLoader.Parameters parameter) {
+		return map;
+	}
+
+	@Override
+	protected Array<AssetDescriptor> getDependencyAssetDescriptors (FileHandle tmjFile,
+		TextureLoader.TextureParameter textureParameter) {
+		Array<AssetDescriptor> descriptors = new Array<>();
+
+		final Array<FileHandle> fileHandles = getDependencyFileHandles(tmjFile);
+		for (FileHandle handle : fileHandles) {
+			descriptors.add(new AssetDescriptor(handle, Texture.class, textureParameter));
+		}
+
+		return descriptors;
+	}
+
+	protected Array<FileHandle> getDependencyFileHandles (FileHandle tmjFile) {
```

Comment:
The Method getDependencyFileHandles() runs into the same issue the TMX loader had. We need to make sure to recursively run through the images which are separated out into the nested group layers. 
But don't fret. I will provide the fixed method and an additional method to check for them recursively:
```
protected Array<FileHandle> getDependencyFileHandles (FileHandle tmjFile) {
		  Array<FileHandle> fileHandles = new Array<>();

		  // TileSet descriptors
		  for (JsonValue tileSet : root.get("tileSets")) {
				getTileSetDependencyFileHandle(fileHandles, tmjFile, tileSet);
		  }

		  // ImageLayer descriptors
		  collectImageLayerFileHandles(root.get("layers"), tmjFile, fileHandles);

		  return fileHandles;
	 }

	 private void collectImageLayerFileHandles(JsonValue layers, FileHandle tmjFile, Array<FileHandle> fileHandles) {
		  if (layers == null) return;

		  for (JsonValue layer : layers) {
				String type = layer.getString("type", "");
				if (type.equals("imagelayer")) {
					 String source = layer.getString("image", null);

					 if (source != null) {
						  FileHandle handle = getRelativeFileHandle(tmjFile, source);
						  fileHandles.add(handle);
					 }
				} else if (type.equals("group")) {
					 // Recursively process group layers
					 collectImageLayerFileHandles(layer.get("layers"), tmjFile, fileHandles);
				}

		  }
	 }
```